Core/src/org/sleuthkit/autopsy/datasourceprocessors/xry/XRYDataSourceProcessor.java

```diff
@@ -0,0 +1,243 @@
+            swingWorker.cancel(true);
+        }
+    }
+
+    @Override
+    public void reset() {
+        //Clear the current selected file path.
+        configPanel.clearSelectedFilePath();
+    }
+
+    /**
+     * Relieves the EDT from add images to the case database and processing the
+     * XRY report files.
+     */
+    private class XRYReportProcessorSwingWorker extends SwingWorker<LocalFilesDataSource, Void> {
+
+        private final DataSourceProcessorProgressMonitor progressMonitor;
+        private final DataSourceProcessorCallback callback;
+        private final FileManager fileManager;
+        private final XRYFolder xryFolder;
+
+        public XRYReportProcessorSwingWorker(XRYFolder folder, DataSourceProcessorProgressMonitor progressMonitor,
+                DataSourceProcessorCallback callback, FileManager fileManager) {
+            this.xryFolder = folder;
+            this.progressMonitor = progressMonitor;
+            this.callback = callback;
+            this.fileManager = fileManager;
+        }
+
+        @Override
+        @NbBundle.Messages({
+            "XRYDataSourceProcessor.preppingFiles=Preparing to add files to the case database",
+            "XRYDataSourceProcessor.processingFiles=Processing all XRY files..."
+        })
+        protected LocalFilesDataSource doInBackground() throws TskCoreException, 
+                TskDataException, IOException {
+            progressMonitor.setProgressText(Bundle.XRYDataSourceProcessor_preppingFiles());
+            
+            List<Path> nonXRYFiles = xryFolder.getNonXRYFiles();
+            List<String> filePaths = nonXRYFiles.stream()
+                    //Map paths to string representations.
+                    .map(Path::toString)
+                    .collect(Collectors.toList());
+            String uniqueUUID = UUID.randomUUID().toString();
+            LocalFilesDataSource dataSource = fileManager.addLocalFilesDataSource(
+                    uniqueUUID, 
+                    "XRY Report", //Name
+                    "", //Timezone
+                    filePaths, 
+                    new ProgressMonitorAdapter(progressMonitor));
+
+            //Process the report files.
+            progressMonitor.setProgressText(Bundle.XRYDataSourceProcessor_processingFiles());
+            XRYReportProcessor.process(xryFolder, dataSource);
+            return dataSource;
+        }
+   
+        @Override
+        @NbBundle.Messages({
+            "XRYDataSourceProcessor.unexpectedError=Internal error occurred while processing XRY report"
+        })
+        public void done() {
+            try {
+                LocalFilesDataSource newDataSource = get();
+                callback.done(DataSourceProcessorCallback.DataSourceProcessorResult.NO_ERRORS, 
+                        Lists.newArrayList(), Lists.newArrayList(newDataSource));
+            } catch (InterruptedException ex) {
+                //DSP was cancelled. Not an error.
```

Comment:
It would probably be a good idea to log a warning here, since there may be incomplete data in the case now.